logsmanagement/rrd_api/rrd_api.h

```diff
@@ -0,0 +1,313 @@
+/** @file rrd_api.h
+ */
+
+#ifndef RRD_API_H_
+#define RRD_API_H_
+
+#include "daemon/common.h"
+
+#include "../circular_buffer.h"
+
+struct Chart_meta;
+struct Chart_str {
+    const char *type;
+    const char *id;    
+    const char *title;
+    const char *units; 
+    const char *family;  
+    const char *context;
+    const char *chart_type; 
+    long priority;
+    int update_every;
+};
+
+#include "rrd_api_generic.h"
+#include "rrd_api_web_log.h"
+#include "rrd_api_kernel.h"
+#include "rrd_api_systemd.h"
+#include "rrd_api_docker_ev.h"
+#include "rrd_api_mqtt.h"
+
+#define LOGS_MANAGEMENT_PLUGIN_STR          "logsmanagement.plugin"
+
+#define CHART_TITLE_TOTAL_COLLECTED_LOGS    "Total collected log records"
+#define CHART_TITLE_RATE_COLLECTED_LOGS     "Rate of collected log records"
+#define NETDATA_CHART_PRIO_LOGS_INCR        100     /**< PRIO increment step from one log source to another **/
+
+typedef struct Chart_data_cus {
+    char *id;
+
+    struct chart_data_cus_dim {
+        char *name;
+        collected_number val;
+        unsigned long long *p_counter;
+    } *dims;
+
+    int dims_size;
+
+    struct Chart_data_cus *next;
+
+} Chart_data_cus_t ;
+
+struct Chart_meta {
+    enum log_src_type_t type;
+    long base_prio;
+
+    union {
+        chart_data_generic_t    *chart_data_generic;
+        chart_data_web_log_t    *chart_data_web_log;
+        chart_data_kernel_t     *chart_data_kernel;
+        chart_data_systemd_t    *chart_data_systemd;
+        chart_data_docker_ev_t  *chart_data_docker_ev;
+        chart_data_mqtt_t       *chart_data_mqtt;
+    };
+
+    Chart_data_cus_t *chart_data_cus_arr;
+
+    void (*init)(struct File_info *p_file_info);
+    void (*update)(struct File_info *p_file_info);
+
+};
+
+static inline struct Chart_str create_chart(const char *type,  
+                                            const char *id,      
+                                            const char *title,
+                                            const char *units, 
+                                            const char *family,  
+                                            const char *context,
+                                            const char *chart_type, 
+                                            long priority,  
+                                            int update_every){
+
```

Comment:
Considering that you are defining a local structure, I suggest making it global, and do not pass a lot of arguments to the function, this will avoid using the stack.